Source/CombatExtended/CombatExtended/Things/Building_TurretGunCE.cs

```diff
@@ -643,6 +643,7 @@ private void InitGun()
                 CompAmmo.turret = this;
                 //if (def.building.turretShellDef != null && def.building.turretShellDef is AmmoDef) CompAmmo.selectedAmmo = (AmmoDef)def.building.turretShellDef;
             }
+            gunInt = GetOrCreateGunInt();
             List<Verb> allVerbs = this.gunInt.TryGetComp<CompEquippable>().AllVerbs;
```

Comment:
Can't you replace `this.gunInt` with `Gun` and let the property handle instancing gunInt? That way you won't need GetOrCreateGunInt() anymore, which seems a bit redundant.

Reply:
The problem is that `Gun` has a bunch of other side effects that I'm not familiar with.